src/content_cao.cpp

```diff
@@ -1492,7 +1492,14 @@ void GenericCAO::updateAnimation()
 
 	if (m_animated_meshnode->getStartFrame() != m_animation_range.X ||
 		m_animated_meshnode->getEndFrame() != m_animation_range.Y)
+                {
+                        f32 current_frame = m_animated_meshnode->getFrameNr();
 			m_animated_meshnode->setFrameLoop(m_animation_range.X, m_animation_range.Y);
+                        // setFrameLoop method has a side effect, it resets the current frame of the animation
+                        if (current_frame > m_animated_meshnode->getStartFrame() &&
+                                current_frame < m_animated_meshnode->getEndFrame())
+                                        m_animated_meshnode->setCurrentFrame(current_frame);
```

Comment:
Intend here the other way -> Two tabs for the new `if` line, one for the function call.
http://dev.minetest.net/Code_style_guidelines#Spaces (see `for` loop example)